src/cdk/scripts/README.md

```diff
@@ -0,0 +1,194 @@
+
+1. **Automatic Execution**: Runs after all CDK stacks are deployed
+2. **Multi-Region Detection**: Automatically includes us-east-1 when WAF is enabled
+3. **S3 Upload**: Uploads dashboard to assets bucket with CloudFront distribution
+4. **URL Generation**: CloudFormation template exposes dashboard URL as output
+
+### Environment Variables
+
+The script uses these environment variables for configuration:
+
+- `AWS_REGION`: Primary region for export scanning
+- `CUSTOM_ENABLE_WAF`: Set to 'true' to include us-east-1 region
+- `ASSETS_BUCKET_NAME`: S3 bucket for dashboard upload (auto-detected from exports)
+
+## File Structure
+
+```
+src/cdk/scripts/
+├── manage-exports.py              # Main script with CLI interface
+├── templates/
+│   └── exports-dashboard.j2       # Professional Jinja2 HTML template
+└── README.md                      # This documentation
+```
+
+## Template Customization
+
+The HTML template can be customized by modifying `templates/exports-dashboard.j2`:
+
+- **Styling**: Update CSS variables for colors and branding
+- **Layout**: Modify HTML structure and component arrangement
+- **Functionality**: Enhance JavaScript for additional features
+- **Content**: Customize text, disclaimers, and help information
+
+## Integration Details
+
+### CloudFormation Template Enhancement
+
+The solution enhances the existing CloudFormation deployment template:
+
+1. **Lambda Function**: Enhanced `rCDKOutputRetrieverFunction` constructs dashboard URL
+2. **Output Addition**: New `oExportsDashboardUrl` output for easy access
+3. **URL Construction**: Automatically detects CloudFront or falls back to S3 direct URL
+
+### CDK Pipeline Stage
+
+Added as final pipeline wave in `lib/pipeline.ts`:
+
+```typescript
+const exportsDashboardWave = pipeline.addWave('ExportsDashboard');
+const exportsDashboardStep = new CodeBuildStep('GenerateExportsDashboard', {
+    commands: [
+        'pip3 install jinja2 boto3',
+        'python3 scripts/manage-exports.py generate-dashboard --filter-prefix Workshop',
+    ],
+});
+```
+
+## Security & Compliance
+
+- **Workshop Disclaimer**: Prominent notice that content is for educational use only
+- **No Sensitive Data**: Filters out internal AWS/CDK exports by default
+- **Read-Only Access**: Script only reads CloudFormation exports, no modifications
+- **Secure Upload**: Uses proper S3 permissions and HTTPS-only access
+
+## Troubleshooting
+
+### Common Issues
+
+**No exports found**: Check that CDK stacks have been deployed and exports exist
+
+```bash
+aws cloudformation list-exports --region us-east-1
+```
+
+**Permission denied**: Ensure AWS credentials have CloudFormation read access
+
+```bash
+aws sts get-caller-identity
+```
+
+**Template not found**: Verify templates directory exists and contains `exports-dashboard.j2`
+
+```bash
+ls -la scripts/templates/
+```
+
+**S3 upload fails**: Check bucket exists and permissions allow PutObject
+
+```bash
+aws s3 ls s3://your-assets-bucket/
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential S3 bucket sniping vulnerability detected. This rule has identified S3 bucket references that could be vulnerable to bucket sniping attacks. Bucket sniping occurs when an attacker registers an S3 bucket name after finding it referenced in code but not yet created. This can lead to data exposure, malicious content hosting, or service disruption.

Recommendations:
1. Create all referenced S3 buckets immediately
2. Use organization-specific prefixes for bucket names
3. Verify bucket ownership before use
4. Consider using AWS Organizations S3 bucket naming rules

Discovered: your-assets-bucket